### PR TITLE
Xeno Larva burst out of the chest, instead of gibbing

### PR DESCRIPTION
**Pull request by FloFluoro**
## What Does This PR Do
Once an Alien Embryo has fully gestated, a larva will now burst out of the chest of the host instead of gibbing them. Chestbursting inflicts a hefty 300 brute damage to the chest, fractures the chest, and disembowels the chest region (heart and lungs).

This PR does **not** change anything regarding how larva clients are selected; first dchat is notified, and if there are no takers, the host is put into the larva.

Also adds a few comments to better describe Alien Embryo functions.

## Why It's Good For The Game
* Massively reduces object spam (liver/eyeballs/kidney/brain/appendix/ears/leftleg/rightleg/leftarm/rightarm for every single host)
* Embryo hosts were only ever gibbed due to design considerations that don't apply to modern Paradise (see below). Updating Alien Embryo gestation to reflect how people actually treat antags is good

The idea behind this PR was floated years ago (#11111), and rejected by heads for the following reason:

>We want the xeno numbers to increase over time and not just have players waiting around forever in the hopes that they get revived

To be blunt, I don't think this is a realistic concern for Paradise in 2022. Dchat is always chomping at the bit to take self-reproducing antag roles like Terror Spiders and Swarmers, to the point that they're often claimed before the prompt icon even appears on-screen. The same was true when Xenomorphs were run as a midround event, just yesterday. IMO it's _very_ unlikely that there will ever be a case where the pop is so low that no one volunteers to become a xenomorph... and even if that does happen, the larva host is still there as a failsafe to make sure all human-gestated larvas have clients.

Xenomorphs have existed in a warped state of balance for years now, which is the reason why they were removed from rotation in #13104 and why subsequent attempts to return them to rotation (#13292 and #13796) have failed. Larva gibbing making it a massive pain to revive recovered victims is, IMO, one of the more glaring issues plaguing Xenos. If the rationale behind gibbing no longer applies to this server, then it should be removed to get Xenomorphs a little closer inline with other antags.

tl;dr if White Terrors don't need to gib to secure clients, then neither do Xenos

## Testing
Compiled and tested on local debug server

## Changelog
:cl:
tweak: Xenomorph Larva now burst out of the victim's chest, instead of gibbing them
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
 
 
-/obj/item/organ/internal/body_egg/alien_embryo/proc/AttemptGrow(gib_on_success = 1)
+/obj/item/organ/internal/body_egg/alien_embryo/proc/AttemptGrow(burst_on_success = 1)
 	if(!owner || polling)
 		return
 	polling = 1
```
Comment (Miraviel) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:71]: Just for simple code readability:

```suggestion
	polling = TRUE
```

Please change `polling = 0` to `FALSE` later, GitHub doesn't let me comment on it.

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 2]
```diff
@@ -91,6 +91,7 @@
 		owner.overlays += overlay
 
 		spawn(6)
```
Comment (Miraviel) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:93]: There are some `spawn()` in this proc, you may want to consider refactoring it, to slowly make xenomorph code up-to-date?

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 3]
```diff
@@ -101,11 +102,13 @@
 			new_xeno << sound('sound/voice/hiss5.ogg',0,0,0,100)//To get the player's attention
```
Comment (Miraviel) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:102]: I think we currently send sounds this way instead:

```suggestion
			SEND_SOUND(new_xeno, sound('sound/voice/hiss5.ogg'))
```

**Review by Miraviel — CHANGES_REQUESTED** [review]
Hey, feel free to dismiss this review. Just pointing out some places where this code could use some refactoring - I think your goal is to get xenomorphs up-to-date, so I thought the code could use some update as well. :)

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 2 above)
Comment (Miraviel) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm:93]: Disregard this, it'd need almost the entire file to be refactored, which I will look into later :>

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 1 above)
Comment (FloFluoro) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Done!

**Comment by Leanfrog** [comment]
image

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 4]
```diff
@@ -84,28 +84,31 @@
...
 			to_chat(new_xeno, "<span class='motd'>For more information, check the wiki page: ([GLOB.configuration.url.wiki_url]/index.php/Xenomorph)</span>")
 
-			if(gib_on_success)
-				owner.gib()
-			else
+			if(burst_on_success) //If we burst naturally
+				owner.apply_damage(300, BRUTE, "chest")
+				owner.bleed(BLOOD_VOLUME_NORMAL)
+				var/obj/item/organ/external/chestorgans = owner.get_organ(ran_zone("chest"))
+				chestorgans.droplimb()
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:109]: this needs to explicitly break their chest. and, it shouldn't be random, and we have defines for these.
```suggestion
			if(burst_on_success) //If we burst naturally
				owner.apply_damage(300, BRUTE, "chest")
				owner.bleed(BLOOD_VOLUME_NORMAL)
				var/obj/item/organ/external/chestorgans = owner.get_organ(BODY_ZONE_CHEST)
				chestorgans.droplimb()
```

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 4 above)
Comment (FloFluoro) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Done and done.

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 5]
```diff
@@ -84,28 +84,32 @@
...
 			stage = 2 // Let's try again later.
-			polling = 0
+			polling = FALSE
 			return
 
 		var/overlay = image('icons/mob/alien.dmi', loc = owner, icon_state = "burst_lie")
 		owner.overlays += overlay
 
 		spawn(6)
+			owner.overlays -= overlay
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:94]: dont remove overlays directly. use `cut_overlay()`

Comment (FloFluoro) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Done.

**Review by hal9000PR — APPROVED** [review]
